pkg/providers/instance/instance.go

```diff
@@ -189,6 +177,23 @@ func (p *Provider) Delete(ctx context.Context, id string) error {
 	return nil
 }
 
+func (p *Provider) CreateTags(ctx context.Context, id string, tags map[string]string) error {
+	ec2Tags := lo.MapToSlice(tags, func(key, value string) *ec2.Tag {
+		return &ec2.Tag{Key: aws.String(key), Value: aws.String(value)}
+	})
+	_, err := p.ec2api.CreateTagsWithContext(ctx, &ec2.CreateTagsInput{
+		Resources: aws.StringSlice([]string{id}),
+		Tags:      ec2Tags,
+	})
+	if err != nil {
+		if awserrors.IsNotFound(err) {
+			return cloudprovider.NewNodeClaimNotFoundError(fmt.Errorf("tagging resource, %w", err))
+		}
+		return fmt.Errorf("tagging resource, %w", err)
```

Comment:
should it be "tagging instance" since we know it's an instance each time? 
```suggestion
			return cloudprovider.NewNodeClaimNotFoundError(fmt.Errorf("tagging instance, %w", err))
		}
		return fmt.Errorf("tagging instance, %w", err)
```